File: buildSrc/src/main/groovy/ru/alexpanchenko/stater/plugin/StateFieldStorage.java
package ru.alexpanchenko.stater.plugin;

import com.android.annotations.NonNull;

import java.util.ArrayList;
import java.util.List;

import ru.alexpanchenko.stater.plugin.model.SaverField;

public final class StateFieldStorage {

  private final ArrayList<SaverField> saverFields = new ArrayList<>();

  public List<SaverField> getFields() {
    return saverFields;
  }

  public void add(@NonNull SaverField field) {
    saverFields.add(field);
  }

  public void clear() {
    saverFields.clear();
  }
}

File: stater/src/main/kotlin/ru/alexpanchenko/stater/plugin/StateFieldStorage.kt
package ru.alexpanchenko.stater.plugin

import ru.alexpanchenko.stater.plugin.model.SaverField
import java.util.*

class StateFieldStorage {

  private val saverFields = ArrayList<SaverField>()

  val fields: List<SaverField> = saverFields

  fun add(field: SaverField) {
    saverFields.add(field)
  }

  fun clear() {
    saverFields.clear()
  }
}